src/bentoml/bentos.py

```diff
@@ -427,76 +427,86 @@ def containerize(bento_tag: Tag | str, **kwargs: t.Any) -> bool:
 @inject
 def serve(
     bento: str,
-    production: bool = False,
-    port: int = Provide[BentoMLContainer.http.port],
-    host: str = Provide[BentoMLContainer.http.host],
     server_type: str = "http",
-    api_workers: int | None = Provide[BentoMLContainer.api_server_workers],
-    backlog: int = Provide[BentoMLContainer.api_server_config.backlog],
     reload: bool = False,
+    production: bool = False,
+    host: str | None = None,
+    port: int | None = None,
     working_dir: str | None = None,
-    ssl_certfile: str | None = None,
-    ssl_keyfile: str | None = None,
-    ssl_ca_certs: str | None = None,
-    # HTTP-specific args
-    ssl_keyfile_password: str | None = None,
-    ssl_version: int | None = None,
-    ssl_cert_reqs: int | None = None,
-    ssl_ciphers: str | None = None,
-    # GRPC-specific args
+    api_workers: int | None = Provide[BentoMLContainer.api_server_workers],
+    backlog: int = Provide[BentoMLContainer.api_server_config.backlog],
+    ssl_certfile: str | None = Provide[BentoMLContainer.ssl.certfile],
+    ssl_keyfile: str | None = Provide[BentoMLContainer.ssl.keyfile],
+    ssl_keyfile_password: str | None = Provide[BentoMLContainer.ssl.keyfile_password],
+    ssl_version: int | None = Provide[BentoMLContainer.ssl.version],
+    ssl_cert_reqs: int | None = Provide[BentoMLContainer.ssl.cert_reqs],
+    ssl_ca_certs: str | None = Provide[BentoMLContainer.ssl.ca_certs],
+    ssl_ciphers: str | None = Provide[BentoMLContainer.ssl.ciphers],
     enable_reflection: bool = Provide[BentoMLContainer.grpc.reflection.enabled],
     enable_channelz: bool = Provide[BentoMLContainer.grpc.channelz.enabled],
     max_concurrent_streams: int
     | None = Provide[BentoMLContainer.grpc.max_concurrent_streams],
-) -> Server:
-    """Launch a BentoServer and returns a client that exposes all APIs defined in target service"""
+) -> ServerHandle:
+    from .serve import construct_ssl_args
+    from ._internal.server.server import ServerHandle
 
-    if server_type.lower() not in ["http", "grpc"]:
+    server_type = server_type.lower()
+    if server_type not in ["http", "grpc"]:
         raise ValueError('Server type must either be "http" or "grpc"')
 
-    args = [
+    ssl_args = {
+        "ssl_certfile": ssl_certfile,
+        "ssl_keyfile": ssl_keyfile,
+        "ssl_ca_certs": ssl_ca_certs,
+    }
+    if server_type == "http":
+        serve_cmd = "serve-http"
+        if host is None:
+            host = BentoMLContainer.http.host.get()
+        if port is None:
+            port = BentoMLContainer.http.port.get()
+            ssl_args.update(
+                ssl_keyfile_password=ssl_keyfile_password,
+                ssl_version=ssl_version,
+                ssl_cert_reqs=ssl_cert_reqs,
+                ssl_ciphers=ssl_ciphers,
+            )
```

Comment:
Er, why is this inside `if port is None`?

Reply:
😮‍💨 oopsie